From c1104234da2d5d98e9bba2a651cce379fab70ab6 Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Fri, 7 Aug 2015 12:59:44 +0800
Subject: [PATCH] MLK-11294 ARM: imx: Resolve conflictions between A7 and M4

Resolve the module usage conflictions between A7/Linux
and M4/FreeRTOS on iMX7D platforms

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts | 16 ++++++++++++++++
 arch/arm/boot/dts/imx7d-sdb-m4.dts               | 16 ++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
index 3ff774f8..46b359f 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
@@ -11,6 +11,10 @@
 	status = "disabled";
 };
 
+&i2c1 {
+	status = "disabled";
+};
+
 &iomuxc {
 	imx7d-12x12-lpddr3-arm2 {
 		pinctrl_uart2_1: uart2grp-1 {
@@ -22,6 +26,14 @@
 	};
 };
 
+&gpt3 {
+	status = "disabled";
+};
+
+&gpt4 {
+	status = "disabled";
+};
+
 &mcctest{
 	status = "okay";
 };
@@ -41,3 +53,7 @@
 	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
 	status = "okay";
 };
+
+&wdog3{
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx7d-sdb-m4.dts b/arch/arm/boot/dts/imx7d-sdb-m4.dts
index 7b57979..d0bb049 100644
--- a/arch/arm/boot/dts/imx7d-sdb-m4.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-m4.dts
@@ -27,6 +27,10 @@
 	status = "disabled";
 };
 
+&i2c2 {
+	status = "disabled";
+};
+
 &iomuxc {
 	imx7d-sdb {
 		pinctrl_uart2_1: uart2grp-1 {
@@ -38,6 +42,14 @@
 	};
 };
 
+&gpt3 {
+	status = "disabled";
+};
+
+&gpt4 {
+	status = "disabled";
+};
+
 &mcctest{
 	status = "okay";
 };
@@ -57,3 +69,7 @@
 	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
 	status = "okay";
 };
+
+&wdog3{
+	status = "disabled";
+};
-- 
1.8.0

